crates/transaction-emitter-lib/src/transaction_generator/nft_mint_and_transfer.rs

```diff
@@ -252,51 +203,78 @@ pub fn create_nft_transfer_request(
 
 pub struct NFTMintAndTransferGeneratorCreator {
     txn_factory: TransactionFactory,
-    creator_account: Arc<Mutex<LocalAccount>>,
+    creator_address: AccountAddress,
+    faucet_accounts: Vec<LocalAccount>,
     collection_name: Vec<u8>,
     token_name: Vec<u8>,
-    rest_client: RestClient,
 }
 
 impl NFTMintAndTransferGeneratorCreator {
     pub async fn new(
-        mut rng: StdRng,
         txn_factory: TransactionFactory,
         root_account: &mut LocalAccount,
-        rest_client: RestClient,
+        txn_executor: &dyn TransactionExecutor,
+        num_workers: usize,
     ) -> Self {
+        let mut rng = StdRng::from_seed(OsRng.gen());
         let mut creator_account = LocalAccount::generate(&mut rng);
+        let creator_address = creator_account.address();
         let collection_name = "collection name".to_owned().into_bytes();
         let token_name = "token name".to_owned().into_bytes();
         initialize_nft_collection(
-            &rest_client,
+            txn_executor,
             root_account,
             &mut creator_account,
             &txn_factory,
             &collection_name,
             &token_name,
         )
         .await;
+
+        let mut faucet_accounts = Vec::new();
```

Comment:
nit: the name `faucet_accounts` is confusing as this is not the `faucet` - may be rename this to `seed_accounts` or `distribution_accounts` ?